src/controller.js

```diff
@@ -1,58 +1,67 @@
-const service = require('./service');
+const _ = require('lodash');
+const listenerService = require('./listenerService');
+const requestService = require('./requestService');
 const controller = {};
 
 controller.getListener = (req, res) => {
-	const listener = service.getListener(req.params.port);
-	res.json(listener);
+	const listener = listenerService.getListener(req.params.port);
+	res.json(listener.toString());
 };
 
-controller.addListener = (req, res) => {
-	service.addListener(req.params.port);
-	res.sendStatus(200);
-};
-
-controller.removeListener = (req, res) => {
-	service.removeListener(req.params.port);
-	res.sendStatus(200);
-};
-
-controller.addRoute = (req, res) => {
+controller.addListenerOrRoute = (req, res) => {
 	const port = req.params.port;
 	const options = req.body;
 
-	if (!options.method) {
-		throw new Error('"method" required');
-	}
+	if (_.isEmpty(options)) {
+		listenerService.addListener(port);
+	} else {
+		if (!options.method) {
+			throw new Error(`"method" required. port=${port} uri=${options.uri}`);
+		}
 
-	if (!options.route) {
-		throw new Error('"route" required');
+		if (!options.uri) {
+			throw new Error(`"uri" required. port=${port}`);
+		}
+
+		listenerService.addRoute(port, options);
 	}
 
-	service.addRoute(port, options);
+	res.sendStatus(200);
+};
+
+controller.removeListener = (req, res) => {
+	listenerService.remove(req.params.port);
 	res.sendStatus(200);
 };
 
 controller.sendChunk = (req, res) => {
 	const port = req.params.port;
-	const route = req.body.route;
+	const uri = req.body.uri;
 	const chunk = Buffer.from(req.body.data, 'base64');
 
-	if (!route) {
-		throw new Error('"route" required');
+	if (!uri) {
+		throw new Error('"uri" required');
 	}
 
-	service.sendChunk(port, route, chunk);
+	listenerService.sendChunk(port, uri, chunk);
```

Comment:
Sorry if I was unclear. I would like to see the logic separated, so that listenerService does not know about chunks. But on the other hand, we already made it to know about `addRoute` logic. 
I think you can leave it as is